state-chain/pallets/cf-lp/src/lib.rs

```diff
@@ -292,12 +292,11 @@ pub mod pallet {
 		#[pallet::call_index(5)]
 		#[pallet::weight(T::WeightInfo::deregister_lp_account())]
 		pub fn deregister_lp_account(who: OriginFor<T>) -> DispatchResult {
-			const STABLE_ASSET: Asset = Asset::Usdc;
 			let account_id = T::AccountRoleRegistry::ensure_liquidity_provider(who)?;
 
 			ensure!(
-				Asset::all().filter(|asset| *asset != STABLE_ASSET).all(|asset| {
-					T::PoolApi::open_order_count(&account_id, asset, STABLE_ASSET)
+				T::PoolApi::pools().iter().all(|asset_pair| {
+					T::PoolApi::open_order_count(&account_id, asset_pair.base, asset_pair.quote)
```

Comment:
why do we even separate base and quote here? out of scope for this PR, but maybe it  makes sense to pass asset_pair through a bit deeper - just reduces number of params, a little tidier, and means there are fewer places to mess up the order of passing in base/quote in the wrong place